src/main/java/ee/tuleva/onboarding/mandate/MandateController.java

```diff
@@ -77,33 +74,33 @@ public MobileSignatureResponse startMobileIdSignature(
       @PathVariable("id") Long mandateId,
       @ApiIgnore @AuthenticationPrincipal AuthenticatedPerson authenticatedPerson) {
 
-    Optional<MobileIDSession> session = genericSessionStore.get(MobileIDSession.class);
+    Optional<MobileIDSession> session = sessionStore.get(MobileIDSession.class);
     MobileIDSession loginSession = session.orElseThrow(IdSessionException::mobileSessionNotFound);
 
     MobileIdSignatureSession signatureSession =
         mandateService.mobileIdSign(
             mandateId, authenticatedPerson.getUserId(), loginSession.getPhoneNumber());
-    genericSessionStore.save(signatureSession);
+    sessionStore.save(signatureSession);
 
-    return new MobileSignatureResponse(signatureSession.challenge, signatureSession.challenge);
+    return new MobileSignatureResponse(signatureSession.challenge);
   }
 
   @ApiOperation(value = "Is mandate successfully signed with mobile ID")
   @RequestMapping(method = GET, value = "/{id}/signature/mobileId/status")
-  public MandateSignatureStatusResponse getMobileIdSignatureStatus(
+  public MobileSignatureStatusResponse getMobileIdSignatureStatus(
       @PathVariable("id") Long mandateId,
       @ApiIgnore @AuthenticationPrincipal AuthenticatedPerson authenticatedPerson) {
 
     Optional<MobileIdSignatureSession> signatureSession =
-        genericSessionStore.get(MobileIdSignatureSession.class);
+        sessionStore.get(MobileIdSignatureSession.class);
     MobileIdSignatureSession session =
         signatureSession.orElseThrow(IdSessionException::mobileSignatureSessionNotFound);
 
     String statusCode =
         mandateService.finalizeMobileIdSignature(
             authenticatedPerson.getUserId(), mandateId, session);
 
-    return new MandateSignatureStatusResponse(statusCode);
+    return new MobileSignatureStatusResponse(statusCode, session.challenge);
```

Comment:
I know it's from a library but is there a reason there is no getter for this field?

Reply:
There is. The `@Data` annotation adds getters.